pkg/models/plan.go

```diff
@@ -1,32 +1,46 @@
 package models
 
 import (
-	"github.com/bacalhau-project/bacalhau/pkg/jobstore"
 	"github.com/bacalhau-project/bacalhau/pkg/model"
 )
 
+type PlanExecutionDesiredUpdate struct {
+	Execution    *model.ExecutionState
+	DesiredState model.ExecutionDesiredState
+	Comment      string
+}
+
 // Plan holds actions as a result of processing an evaluation by the scheduler.
 type Plan struct {
 	EvalID      string
 	EvalReceipt string
-	Priority    int
+	// TODO: passing the evalID should be enough once we persist evaluations
+	Eval     *Evaluation
+	Priority int
+
+	Job        *model.Job
+	JobVersion int
+
+	DesiredJobState model.JobStateType
+	Comment         string
 
 	// NewExecutions holds the executions to be created.
 	NewExecutions []*model.ExecutionState
 
-	// StoppedExecutions holds the executions to be stopped.
-	StoppedExecutions []*jobstore.UpdateExecutionRequest
+	UpdatedExecutions map[model.ExecutionID]*PlanExecutionDesiredUpdate
 }
 
 // NewPlan creates a new Plan instance.
-func NewPlan(eval *Evaluation) *Plan {
+func NewPlan(eval *Evaluation, job *model.Job, jobVersion int) *Plan {
 	return &Plan{
```

Comment:
As a reader I'd find it helpful to be explicit here about the value of `DesiredState`, which I believe defaults to `JobStateNew`? Can we set that field in the constructor?

Reply:
`DesiredJobState` is optional if the plan must update the job state. The default `JobStateUndefined` means the plan should not update the job state